### DISCUSSION TITLE
Release plan / Release 1.25.5 / 1.27.2

### DISCUSSION THREAD

**Discussion by rborundia** [cri-o/cri-o#7469] [Q&A]
Hi,
Could you please let us know when are we going to get release for build number 1.27.2/1.25.5 ?
We are looking for following fix(PR #7129). Release1.27.1 and 1.25.4 does not have the following fix. We are running to crio panic with double close of channel when we delete a deployment and crio's container stop function goes into retry.

commit https://github.com/cri-o/cri-o/commit/6dec88e7cca04babc2d4eaadd904eb2d6bfe07e4
Author: Peter Hunt [pehunt@redhat.com](mailto:pehunt@redhat.com)
Date: Thu Jul 6 16:58:49 2023 -0400

oci: simplify stopping code

Before, there was the possibility load could cause cri-o to segfault from double closing of channels.

this PR aims to simplify container stop code while retaining the required behavior.

Now, the first stop begins a registration process where the container stop begins and new timeouts
come in to interrupt. There are two commuincation channels, and only one location where they can be closed.
This also adds a watcher mechanism so callers can wait on the container stop

Signed-off-by: Peter Hunt <pehunt@redhat.com>

Panic:
Oct 25 02:18:41 dssserv8 crio[2263]: panic: close of closed channel
Oct 25 02:18:41 dssserv8 crio[2263]: goroutine 2601932 [running]:
Oct 25 02:18:41 dssserv8 crio[2263]: panic({0x5641fb169a00, 0x5641fb420080})
Oct 25 02:18:41 dssserv8 crio[2263]:         /usr/lib64/go-1.19/src/runtime/panic.go:987 +0x3ba fp=0xc0020f3990 sp=0xc0020f38d0 pc=0x5641f92f017a
Oct 25 02:18:41 dssserv8 crio[2263]: runtime.closechan(0xc0018cbf20)
Oct 25 02:18:41 dssserv8 crio[2263]:         /usr/lib64/go-1.19/src/runtime/chan.go:365 +0x3fc fp=0xc0020f39d0 sp=0xc0020f3990 pc=0x5641f92bae5c
Oct 25 02:18:41 dssserv8 crio[2263]: github.com/cri-o/cri-o/internal/oci.WaitContainerStop({0x5641fb439d48?, 0xc0008b72c0?}, 0xc001402580, 0x1bf08eb000, 0>
Oct 25 02:18:41 dssserv8 crio[2263]:         /home/abuild/rpmbuild/BUILD/cri-o-1.27.1/internal/oci/runtime_oci.go:864 +0x671 fp=0xc0020f3b18 sp=0xc0020f39>
Oct 25 02:18:41 dssserv8 crio[2263]: github.com/cri-o/cri-o/internal/oci.(*runtimeOCI).StopContainer(0xc0006b9170?, {0x5641fb439d48?, 0xc0008b7290?}, 0xc0>
Oct 25 02:18:41 dssserv8 crio[2263]:         /home/abuild/rpmbuild/BUILD/cri-o-1.27.1/internal/oci/runtime_oci.go:920 +0x585 fp=0xc0020f3d10 sp=0xc0020f3b>
Oct 25 02:18:41 dssserv8 crio[2263]: github.com/cri-o/cri-o/internal/oci.(*Runtime).StopContainer(0x5641fb439d48?, {0x5641fb439d48?, 0xc0008b71a0?}, 0x0?,>
Oct 25 02:18:41 dssserv8 crio[2263]:         /home/abuild/rpmbuild/BUILD/cri-o-1.27.1/internal/oci/oci.go:284 +0x11c fp=0xc0020f3da8 sp=0xc0020f3d10 pc=0x>
Oct 25 02:18:41 dssserv8 crio[2263]: github.com/cri-o/cri-o/server.(*Server).stopContainer(0xc0000fbc00, {0x5641fb439d48?, 0xc001b9c4e0?}, 0xc001402580, 0>
Oct 25 02:18:41 dssserv8 crio[2263]:         /home/abuild/rpmbuild/BUILD/cri-o-1.27.1/server/container_stop.go:60 +0x43b fp=0xc0020f3f08 sp=0xc0020f3da8 p>
Oct 25 02:18:41 dssserv8 crio[2263]: github.com/cri-o/cri-o/server.(*Server).stopPodSandbox.func1()

**Comment by kwilczynski** [comment]
@rborundia, thank you for asking!

Work on the release **1.27.2** is already ongoing per https://github.com/cri-o/cri-o/pull/7430.

However, I wonder if we have plans to release a new 1.25.x version at this time.

That said, regardless of which release branch, we would need to ensure this bug fix is backported.

@saschagrunert and @haircommander, your thoughts?

**Reply by rborundia** [reply]
Hi Kwilczynski,

Thank you for sharing the info about 1.27.2 release.
@saschagrunert   and @haircommander  could you please share your thoughts on 1.25.x release with above fixes.

Thanks,
Rajesh

**Reply by saschagrunert** [reply]
I'm not against a new release if the community wants one.

**Reply by haircommander** [reply]
yeah we can cut a new 1.25.4, it probably will be one of the last ones though as it's out of upstream support now

**Reply by rborundia** [reply]
Hi Kwilczynski,

Is it possible to know tentative date when release 1.27.2 will be available ?

Thanks,
Rajesh

**Reply by kwilczynski** [reply]
@rborundia,

I will bring this up with the team today. Please stay put for now.

Also, sorry for the delay!

**Reply by haircommander** [reply]
I'm packaging it today!

**Reply by haircommander** [reply]
thanks for your patience

**Reply by rborundia** [reply]
@kwilczynski 
Apologies for late response. No problem, I needed the date so that we can plan accordingly.

@haircommander 
Again apologies for late response. Thanks.

**Reply by rborundia** [reply]
@haircommander Did you get a chance to create package for 1.27.2.

Thanks,
Rajesh

**Reply by haircommander** [reply]
1.27.2 is done now! sorry for the delay, my scripts were broken

**Reply by rborundia** [reply]
@haircommander Thanks, do you know when it will be available as a binary officially. I do not see it in releases yet.

**Reply by haircommander** [reply]
hm @saschagrunert something is wrong with the release note automation: https://github.com/cri-o/cri-o/commit/19bae013072afaef77d86241b2da886607509fdf it seems like we clobbered 1.28.2 release with 1.27.2 bump

**Reply by saschagrunert** [reply]
@haircommander do you mind opening a separate issue for that?

**Reply by haircommander** [reply]
done! https://github.com/cri-o/cri-o/issues/7591

**Reply by trekker-1975** [reply]
I am also running into this same issue.  We are currently on v1.25 and will be moving to v1.26.  When will the releases for v1.25.5 and v1.26.5 be available to pick up this fix?

**Reply by rborundia** [reply]
Hi @haircommander 
Do you know when #7591 will be accepted and when we will get official build. We are waiting for this build for the release.

**Reply by haircommander** [reply]
I have just attempted to cut 1.26.5, 1.25.5 artifacts are broken so we may need to do a 1.25.6 in the new year. sorry for the delay. 1.27.2 should be ready now

**Reply by rborundia** [reply]
@haircommander 
I see that 1.27.2 is released. Thank you for the release. Do you know when binaries will be available at https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.27:/1.27.2/CentOS_8/

**Reply by kwilczynski** [reply]
@rborundia, I might need to be corrected about this concerning older releases. However, the packages repository has moved. 

Check the following blog post:

- [CRI-O is moving towards pkgs.k8s.io](https://kubernetes.io/blog/2023/10/10/cri-o-community-package-infrastructure)

More details can be found at [cri-o/packaging](https://github.com/cri-o/packaging) project.

**Reply by haircommander** [reply]
sorry @rborundia I messed one last thing up. Hopefully it works now. @kwilczynski you were correct about needing to be corrected, we'll remain on the old paths for 1.28 and older :)

**Reply by kwilczynski** [reply]
@haircommander, no worries. I am happy to be wrong about this! :smile:
